### PR TITLE
rt: allow configuring I/O events capacity

### PR DESCRIPTION
**Pull request by littledivy**
Adds a method `Builder::max_io_events_per_tick()` to the runtime builder. This can be used to configure the capacity of events that may be processed per OS `poll`.

### CODE REVIEW

File: tokio/src/runtime/builder.rs [hunk 1]
```diff
@@ -938,6 +941,25 @@ cfg_io_driver! {
...
+        /// ```
+        /// use tokio::runtime;
+        ///
+        /// let rt = runtime::Builder::new_current_thread()
+        ///     .enable_io()
+        ///     .with_events_capacity(1024)
+        ///     .build()
+        ///     .unwrap();
+        /// ```
+        pub fn with_events_capacity(&mut self, capacity: usize) -> &mut Self {
```
Comment (carllerche) [inline tokio/src/runtime/builder.rs]: we don't use `with`, maybe try something like `max_io_events_per_tick` or something. Not perfect, but be descriptive.

File: tokio/src/runtime/builder.rs [hunk 2]
```diff
@@ -938,6 +941,25 @@ cfg_io_driver! {
             self.enable_io = true;
             self
         }
+
+        /// Enables the I/O driver and configures the capacity of events to be
+        /// processed per tick.
```
Comment (Darksonn) [inline tokio/src/runtime/builder.rs:946]: Capacity?

**Comment by Darksonn** [comment]
Hi, did you see my previous comment?

File: tokio/src/runtime/builder.rs (hunk 2 above)
Comment (littledivy) [reply tokio/src/runtime/builder.rs]: Replaced with `max number of events`

**Review by carllerche — APPROVED** [review]
LGTM.

**Review by Noah-Kennedy — APPROVED** [review]

**Pull request merged**